design/secrets_provider_milestone1_design.md

```diff
@@ -0,0 +1,474 @@
+  * [Order of Deployment](#order-of-deployment)
+  * [Backwards compatibility](#backwards-compatibility)
+  * [Performance](#performance)
+  * [Affected Components](#affected-components)
+- [Security](#security)
+- [Test Plan](#test-plan)
+- [Logs](#logs)
+  * [Audit](#audit)
+- [Documentation](#documentation)
+- [Open questions](#open-questions)
+- [Implementation plan](#implementation-plan)
+
+## Glossary
+
+| **Term**                                                     | **Description**                                              |
+| ------------------------------------------------------------ | ------------------------------------------------------------ |
+| [Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/) | K8s entity that runs a pod for one-time operation. Once the pod exists, the Job terminates. |
+| [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) | K8s entity that ensures multiple replications of a pod are running all the time. If a pod terminates, another one will be created. |
+
+## Useful links
+
+| Name                | Link                                                         |
+| ------------------- | ------------------------------------------------------------ |
+| Phase 2 Feature Doc | https://github.com/cyberark/secrets-provider-for-k8s/issues/102 |
+
+## Background
+
+The Cyberark Secrets Provider for K8s enables customers to use secrets stored and managed in Conjur/DAP vault and consume them as K8s Secrets for their application containers.
+
+During Phase 1, an *init container* was deployed in the same Pod as the customer's applications to populate the K8s secrets defined in the pod application's manifest.
+
+### Motivation
+
+The current implementation of the Secrets Provider only supports one application and makes the deployment of the Secrets Provider deeply coupled to the customer's applications.
+
+### Requirements
+
+The solution should support the following capabilities:
+
+Milestone 1 *(current)*
+
+- Secrets Provider runs and deployed without affecting the app deployment or app life cycle
+- Secrets Provider can serve many apps
+
+Milestone 2
+
+- Secrets Provider support secret rotation
+
+### Future requirements
+
+The solution should consider the following future requirements that may arise:
+
+* Support other **targets** for the secrets besides K8s Secrets, such as files
+* Support other **sources** for secrets paths besides conjur-map in K8s Secrets
+* Support **sources** varying while Secrets Provider runs
+
+## Solution
+The solution is to allow different deployments and behave differently based on the use-case chosen.
+
+The following decision tree depicts customer uses-cases for Secrets Provider deployments:
+
+![Secrets Provider flavors desicion flow chart](https://user-images.githubusercontent.com/31179631/85747023-975bf500-b70f-11ea-8e26-1134068fe655.png)
+
+This decision tree shows 6 different variations of Secrets Provider:
```

Comment:
I think we should consider our secret provider solution to be a deployed as one common image for all cases below.
The only differences should be a parameters / arguments provided to it (such as secrets-destination, with-rotation and etc.), and of coarse the manifests

Reply:
That is the way we want to go. See
> These variations will be supported using the same Secrets Provider image, but the behavior will vary dynamically depending on the chosen deployment.